tests/python/relay/aot/test_crt_forward_declarations.py

```diff
@@ -0,0 +1,325 @@
+        (1, 32, 32, 16),
+        (3, 3),
+        16,
+        1,
+        1,
+        (0, 2, 2, 0),
+        1,
+    )
+    in_dtype = "int8"
+    data_layout = "NHWC"
+    kernel_layout = "HWOI"
+    ref_kernel_layout = "HWIO"
+    out_layout = "NHWC"
+    schedule_name = "conv2d_nhwc_dsp.arm_cpu"
+
+    ref_input_data = np.random.randint(low=-128, high=127, size=data_shape, dtype=in_dtype)
+    ref_input_var = relay.var("input", relay.TensorType(data_shape, in_dtype))  # NHWC layout
+    kernel_shape = (*kernel_size, data_shape[-1] // groups, num_filter)  # HWIO layout
+    ref_kernel_data = np.random.randint(low=-10, high=10, size=kernel_shape, dtype=in_dtype)
+
+    ref_relay_op = relay.op.nn.conv2d(
+        ref_input_var,
+        relay.const(_change_ndarray_layout(ref_kernel_data, "HWIO", ref_kernel_layout)),
+        kernel_size=kernel_size,
+        strides=strides,
+        padding=padding,
+        groups=groups,
+        dilation=(dilation, dilation),
+        data_layout="NHWC",
+        kernel_layout=ref_kernel_layout,
+        out_dtype="int32",
+        out_layout="NHWC",
+    )
+    ref_module = tvm.IRModule.from_expr(relay.Function([ref_input_var], ref_relay_op))
+    ref_outputs = generate_ref_data(ref_module, {"input": ref_input_data})
+
+    # Reshape output dictionary to match out_layout
+    assert len(ref_outputs) == 1
+    output_tensor_name, output_tensor = next(iter(ref_outputs.items()))
+    ref_outputs[output_tensor_name] = _change_ndarray_layout(output_tensor, "NHWC", out_layout)
+
+    test_input_data = _change_ndarray_layout(ref_input_data, "NHWC", data_layout)
+    test_input_var = relay.var("input", relay.TensorType(test_input_data.shape, in_dtype))
+    test_kernel_data = _change_ndarray_layout(ref_kernel_data, "HWIO", kernel_layout)
+
+    test_relay_op = relay.op.nn.conv2d(
+        test_input_var,
+        relay.const(test_kernel_data),
+        kernel_size=kernel_size,
+        strides=strides,
+        padding=padding,
+        groups=groups,
+        dilation=(dilation, dilation),
+        data_layout=data_layout,
+        kernel_layout=kernel_layout,
+        out_dtype="int32",
+        out_layout=out_layout,
+    )
+    test_function = relay.Function([test_input_var], test_relay_op)
+    test_model = AOTTestModel(
+        module=tvm.IRModule.from_expr(test_function),
+        inputs={"input": test_input_data},
+        outputs=ref_outputs,
+    )
+    compiled_models = compile_models(
+        test_model,
+        interface_api="c",
+        use_unpacked_api=True,
+        pass_config=AOT_CORSTONE300_RUNNER.pass_config,
+        target=f"c -keys=arm_cpu -mcpu=cortex-m7",
+        schedule_name=schedule_name,
+    )
+
+    lib_mod = compiled_models[0].executor_factory.lib.imported_modules[0]
+    main_source = lib_mod.get_source()
+    assert main_source.count("tvmgen_default_fused_nn_conv2d") == 2
+    assert main_source.count("gemm_") == 13
```

Comment:
We should consider using a test-oriented schedules/BYOC here rather than these tests being dependent on a specific implementation elsewhere in the codebase - this is fine for now though.